frame/example/src/lib.rs

```diff
@@ -496,7 +506,9 @@ pub mod pallet {
 		// calls to be executed - we don't need to care why. Because it's privileged, we can
 		// assume it's a one-off operation and substantial processing/storage/memory can be used
 		// without worrying about gameability or attack scenarios.
-		#[pallet::weight(WeightForSetDummy::<T>(<BalanceOf<T>>::from(100u32)))]
```

Comment:
the documentation for `WeightForSetDummy` is outdated now.

Maybe we can just remove it as we prefer to just show benchmarking way, not sure. cc @kianenigma

Reply:
I am thinking it is a good way to showcasing how to set the weight: 1) thru benchmarking, 2) manually setting it via WeightForSetDummy, and it shows setting an extrinsic call as operational call. 

I think it will be great showing runtime developers alternatives that they can override the measured benchmarking and tweak the weight function as they want. Will there be some scenarios that this is necessary for dev?

Reply:
ideally what people need to understand is that pallet::weight take into argument any type which implements:
* `WeighData<(Arg1, Arg2, ..)>`
* `ClassifyDispatch<(Arg1, Arg2, ..)>`
* `PaysFee<(Arg1, Arg2, ..)>`

where Arg1, Arg2, ..., are the type of the argument of the call.

Make we can keep WeightForSetDummy for this call.

I think if we keep WeightForSetDummy in the code and never use it, people can't understand how to use it.

Reply:
oh boy I totally have forgotten about these manual weights. 

I think they can stay, only if we mention that this is an advanced feature if you want some custom functionality. 

If too much for now, @jimmychu0807 you can remove it in this PR, make a new Issue in the devhub for _custom weight calculators_ and ping me there, I can help with that.

Reply:
@kianenigma We kind of have that in devhub [here](https://substrate.dev/docs/en/knowledgebase/runtime/fees#default-weight-annotations), though we should probably move it to the [Transaction Weight document](https://substrate.dev/docs/en/knowledgebase/learn-substrate/weight). 

And [the example implementation of WeightForSetDummy](https://github.com/paritytech/substrate/blob/jc%2Ftry-benchmarking/frame/example/src/lib.rs#L303-L333) demonstrates what you could customize.